miso.cabal

```diff
@@ -151,17 +151,17 @@ library
     tagsoup,
     text,
     transformers
-  if impl(ghcjs)
+  if impl(ghcjs) || arch(javascript)
     build-depends:
       ghcjs-base
   else
     build-depends:
       servant-lucid
-  if impl(ghcjs) || flag (jsaddle)
+  if impl(ghcjs) || arch(javascript) || flag (jsaddle)
     if flag (ios)
       cpp-options:
         -DIOS
-  if impl(ghcjs)
+  if impl(ghcjs) || arch(javascript)
```

Comment:
Linking should happen even with cabal 3.10 because these `js-sources` are in a library. @hamishmack Which cabal version were you using?